### DISCUSSION TITLE
Have replacement value match github-release name?

### DISCUSSION THREAD

**Discussion by rishid** [renovatebot/renovate#24573] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovatebot/github-action@3cef36a9aba515d8726b491905b3bc766832e221 # v39.0.5 

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want Renovate replacement to use the full release name.

I had this line originally:
`bat: "v0.22.0"               # renovate: depName=sharkdp/bat`

Renovate correctly created a PR but it was to this change:
`bat: "0.23.0"               # renovate: depName=sharkdp/bat`

I want the full release name to be used which is "v0.23.0".

Relevant reg ex manager
```js
{
  $schema: "https://docs.renovatebot.com/renovate-schema.json",
  regexManagers: [
    {
      description: "Process various dependencies in YAML files",
      fileMatch: ["\\.ya?ml$"],
      matchStrings: [
        // Inline
        '\\S+:\\s+"?(?<currentValue>[^"\\s]+)"?\\s*# ?renovate: depName=(?<depName>\\S+)( datasource=(?<datasource>\\S+))?( versioning=(?<versioning>\\S+))?( extractVersion=(?<extractVersion>\\S+))?',
        // Newline
        '(?m:^\\s+# ?renovate: depName=(?<depName>\\S+)( datasource=(?<datasource>\\S+))?( versioning=(?<versioning>\\S+))?( extractVersion=(?<extractVersion>\\S+))?\\n[ \\t ]*? \\S+: "?(?<currentValue>[^" ]+)"?$)',
        // Old style newline
        '(?m:^\\s+# ?renovate: datasource=(?<datasource>\\S+) depName=(?<depName>\\S+)( versioning=(?<versioning>\\S+))?( versionTemplate=(?<extractVersion>\\S+))?\\n[ \\t ]*? \\S+: "?(?<currentValue>[^" ]+)"?$)'
      ],
      datasourceTemplate: "{{#if datasource}}{{{datasource}}}{{else}}github-releases{{/if}}",
      versioningTemplate: "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}",
      extractVersionTemplate: "{{#if extractVersion}}{{{extractVersion}}}{{else}}^v?(?<version>.*)${{/if}}",
    },
  ],
}
```

### What you tried so far.

Reading the docs and playing with `autoReplaceStringTemplate` which I don't think is the right thing to fix this tbh.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
It looks like your `extractVersionTemplate` is stripping the `v`
